sql/core/src/main/scala/org/apache/spark/sql/execution/commands.scala

```diff
@@ -178,3 +180,34 @@ case class DescribeCommand(
     child.output.map(field => Row(field.name, field.dataType.toString, null))
   }
 }
+
+/**
+ * :: DeveloperApi ::
+ */
+@DeveloperApi
+case class DDLDescribeCommand(
+    dbName: Option[String],
+    tableName: String, isExtended: Boolean) extends RunnableCommand {
+
+  override def run(sqlContext: SQLContext) = {
+    val tblRelation = dbName match {
+      case Some(db) => UnresolvedRelation(Seq(db, tableName))
+      case None => UnresolvedRelation(Seq(tableName))
+    }
+    val logicalRelation = sqlContext.executePlan(tblRelation).analyzed
+    val rows = new ArrayBuffer[Row]()
+    rows ++= logicalRelation.schema.fields.map{field =>
+      Row(field.name, field.dataType.toSimpleString, null)}
+
+    /*
+     * TODO if future support partition table, add header below:
+     * # Partition Information
+     * # col_name data_type comment
```

Comment:
after finish SPARK-5182.  we can do it like this:
```
    val logicalRelation = sqlContext.executePlan(tblRelation).analyzed
    val rows = new ArrayBuffer[Row]()
    rows ++= logicalRelation.schema.fields.map{field =>
      Row(field.name, field.dataType.toSimpleString, null)}
	val partitionFields = logicalRelation.schema.getPartitionedCols()
	if (partitionFields.nonEmpty) {
	  val partColumnRows =
        partitionFields.map(field => Row(field.getName, field.getType.toSimpleString, null))
	  rows ++=
          Seq(("# Partition Information", "", "")) ++
		      ("col_name","data_type","comment") ++ 
          partColumnRows
	}
```
For `extended` we can do it later and discuss to show detail information.
```
    if (isExtended) {
      rows ++= Seq(("Detailed Table Information", //get some detail info from table, ""))
    }
```